### PR TITLE
OCM-4302 | fix: hide billingaccount for classic

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/OCM-4302
image
image

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -981,80 +981,82 @@ func run(cmd *cobra.Command, _ []string) {
...
 
-		if isContractEnabled && len(contracts) > 0 {
-			//currently, an AWS account will have only one ROSA HCP active contract at a time
-			contractDisplay := GenerateContractDisplay(contracts[0])
-			r.Reporter.Infof(contractDisplay)
-		}
+	if isHostedCP && cmd.Flags().Changed("default-mp-labels") {
+		r.Reporter.Errorf("Setting the worker machine pool labels is not supported for hosted clusters")
+		os.Exit(1)
 	}
```
Comment (cristianoveiga) [inline cmd/create/cluster/cmd.go:1060]: We are checking `isHostedCP` above. Should this check go in the same block?

Comment (chenz4027) [reply cmd/create/cluster/cmd.go:1060]: Sure, let me move this in

Comment (chenz4027) [reply cmd/create/cluster/cmd.go]: Updated

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -981,79 +981,81 @@ func run(cmd *cobra.Command, _ []string) {
...
+			}
 
-		if isContractEnabled && len(contracts) > 0 {
-			//currently, an AWS account will have only one ROSA HCP active contract at a time
-			contractDisplay := GenerateContractDisplay(contracts[0])
-			r.Reporter.Infof(contractDisplay)
+			if cmd.Flags().Changed("default-mp-labels") {
+				r.Reporter.Errorf("Setting the worker machine pool labels is not supported for hosted clusters")
+				os.Exit(1)
+			}
```
Comment (cristianoveiga) [inline cmd/create/cluster/cmd.go:1058]: Now this only being checked for the interactive mode, probably not what you wanted. Can you please take a look?

Comment (chenz4027) [reply cmd/create/cluster/cmd.go]: Ah thanks for catching that! Moved it out

File: cmd/create/cluster/cmd.go [hunk 3]
```diff
@@ -981,79 +981,80 @@ func run(cmd *cobra.Command, _ []string) {
...
-
-	if billingAccount != "" && !ocm.IsValidAWSAccount(billingAccount) {
-		r.Reporter.Errorf("Billing account is invalid. Run the command again with a valid billing account.")
-		os.Exit(1)
-	}
+	if isHostedCP {
+		if billingAccount != "" {
+			r.Reporter.Errorf("Billing accounts are only supported for Hosted Control Plane clusters")
+			os.Exit(1)
+		}
```
Comment (cristianoveiga) [inline cmd/create/cluster/cmd.go:988]: @chenz4027 can you please review this logic? It's currently not accepting billing accounts for hosted control plane clusters.

Comment (chenz4027) [reply cmd/create/cluster/cmd.go]: I moved it to an else statement, thanks for catching the flaw: 
image

**Review by cristianoveiga — APPROVED** [review]
LGTM

**Comment by chenz4027** [comment]
@svetsa-rh Sure, unfortunately we don't have permissions to merge. Tagging @ciaranRoche @gdbranco for review

**Comment by krishvoor** [comment]
@chenz4027 @ciaranRoche @gdbranco  We are impacted by this severely our testing is failing and PRs are blocked.
Kindly help merge.

**Pull request merged**